ui/app/js/controllers/required_signoffs_controller.js

```diff
@@ -43,6 +43,33 @@ function($scope, $modal, $q, ProductRequiredSignoffs, PermissionsRequiredSignoff
     }
   }, true);
 
+  $scope.undoRemoveRole = function(role, rs, csrf_token) {
+    var roleName = role.$key;
```

Comment:
Although $key is accessible here, it's not safe to refer to it. You should pass it in as another argument instead, if you truly need it.